ext/RMagick/rmilist.c

```diff
@@ -1176,15 +1248,22 @@ ImageList_write(VALUE self, VALUE file)
     for (img = images; img; img = GetNextImageInList(img))
     {
         rm_sync_image_options(img, info);
+#if defined(IMAGEMAGICK_7)
+        (void) WriteImage(info, img, exception);
+        rm_check_exception(exception, img, RetainOnError);
```

Comment:
With `RetainOnError`, it will cause memory leak if raise an exception.

```suggestion
        rm_check_exception(exception, img, DestroyOnError);
```